transformer/dockerfilegenerator/java/jaranalyser.go

```diff
@@ -78,158 +76,171 @@ func (t *JarAnalyser) GetConfig() (transformertypes.Transformer, *environment.En
 }
 
 // DirectoryDetect runs detect in each sub directory
-func (t *JarAnalyser) DirectoryDetect(dir string) (services map[string][]transformertypes.Artifact, err error) {
-	services = map[string][]transformertypes.Artifact{}
-	jarFilePaths, err := common.GetFilesInCurrentDirectory(dir, nil, []string{".*[.]jar"})
+func (t *JarAnalyser) DirectoryDetect(dir string) (map[string][]transformertypes.Artifact, error) {
+	services := map[string][]transformertypes.Artifact{}
+	paths, err := common.GetFilesInCurrentDirectory(dir, nil, []string{`.*\.jar$`})
 	if err != nil {
-		logrus.Errorf("Error while parsing directory %s for jar file : %s", dir, err)
-		return nil, err
+		return nil, fmt.Errorf("failed to get the jar files in the directory %s . Error: %q", dir, err)
 	}
-	if len(jarFilePaths) == 0 {
+	if len(paths) == 0 {
 		return nil, nil
 	}
-	for _, path := range jarFilePaths {
-		envVariablesMap := map[string]string{}
-		envVariablesMap["PORT"] = fmt.Sprintf("%d", t.JarConfig.DefaultPort)
+	for _, path := range paths {
+		relPath, err := filepath.Rel(t.Env.GetEnvironmentSource(), path)
+		if err != nil {
+			logrus.Errorf("failed to make the path %s relative to the sourc code directory %s . Error: %q", path, t.Env.GetEnvironmentSource(), err)
+			continue
+		}
 		newArtifact := transformertypes.Artifact{
 			Paths: map[transformertypes.PathType][]string{
 				artifacts.JarPathType:        {path},
-				artifacts.ServiceDirPathType: {filepath.Dir(path)},
+				artifacts.ServiceDirPathType: {dir},
 			},
 			Configs: map[transformertypes.ConfigType]interface{}{
 				artifacts.JarConfigType: artifacts.JarArtifactConfig{
-					DeploymentFile: filepath.Base(path),
-					EnvVariables:   envVariablesMap,
-					Port:           t.JarConfig.DefaultPort,
-					JavaVersion:    t.JarConfig.JavaVersion,
+					DeploymentFilePath: relPath,
```

Comment:
Why do we need path relative to source directory in this config? the paths in the artifact already have the right paths to place the Dockerfile. In here the paths are relative to the paths in the Paths section of the artifact, in this case, its just the archive name

Reply:
see above

Reply:
We seem to be computing the path relative to input directory and not service directory.

Reply:
done